common/datavalues/src/types/serializations/number.rs

```diff
@@ -38,12 +38,14 @@ impl<T: PrimitiveType> Default for NumberSerializer<T> {
 
 impl<T> TypeSerializer for NumberSerializer<T>
 where T: PrimitiveType
-        + common_clickhouse_srv::types::StatBuffer
+        + opensrv_clickhouse::types::StatBuffer
         + Marshal
         + Unmarshal<T>
         + HasSqlType
-        + std::convert::Into<common_clickhouse_srv::types::Value>
-        + std::convert::From<common_clickhouse_srv::types::Value>
+        + std::convert::Into<opensrv_clickhouse::types::Value>
+        + std::convert::From<opensrv_clickhouse::types::Value>
+        + opensrv_clickhouse::io::Marshal
+        + opensrv_clickhouse::io::Unmarshal<T>
```

Comment:
The only problem is that we have to support 

```
        + opensrv_clickhouse::io::Marshal
        + opensrv_clickhouse::io::Unmarshal<T>
```

And the same code(Marshal/Unmarshal) is maintained in both common-io and opensrv-clickhouse. Maybe we should have a single crate to do this work.

Reply:
Yes, this is a little weird...

Reply:
Indeed!

Reply:
I think we can move Marshal&Unmarshal to the databend/common, datavalues is a very common crate, depends on a opensrv is very weird :)

Reply:
> we can move Marshal&Unmarshal to the databend/common

common-io does implement the Marshal&Unmarshal, But opensrv-clickhouse also requires Marshal&Unmarsha.

So it would be nice to have a simple standalone crate to do this.